kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/RecordEncryptionFilter.java

```diff
@@ -139,6 +154,19 @@ private CompletionStage<ProduceRequestData> maybeEncodeProduce(ProduceRequestDat
                 });
     }
 
+    private void generatePlainRecordsMetrics(Meter.MeterProvider<Counter> plainRecordsTotal, Set<String> unresolvedTopicNames,
+                                             Map<String, TopicProduceData> topicNameToData) {
+        unresolvedTopicNames.forEach(unresolvedTopic -> {
+            TopicProduceData data = topicNameToData.get(unresolvedTopic);
+            if (data != null) {
+                data.partitionData()
+                        .forEach(produceData -> plainRecordsTotal
+                                .withTags(RecordEncryptionMetrics.TOPIC_NAME, unresolvedTopic)
+                                .increment(RecordEncryptionUtil.totalRecordsInBatches((MemoryRecords) produceData.records())));
+            }
+        });
```

Comment:
In reply to https://github.com/kroxylicious/kroxylicious/pull/2308/files#r2152809984 How about:
```suggestion
                topicNameToData.entrySet().stream()
                .filter(topicDataEntry -> unresolvedTopicNames.contains(topicDataEntry.getKey()))
                .forEach(topicData -> topicData.getValue().partitionData()
                .forEach(produceData -> plainRecordsTotal
                        .withTags(RecordEncryptionMetrics.TOPIC_NAME, topicData.getKey())
                        .increment(RecordEncryptionUtil.totalRecordsInBatches((MemoryRecords) produceData.records()))));
```